boards: infineon: psc3m5_evk: Add support for PSC3M5_EVK

sreeramIfx opened the pull request:
* Adding folder structure and first pass config for PCS3M5_EVK
* Adding dtsi, kconfig, and other files for PSOC Control C3 devices
      and KIT_PSC3M5_EVK evaluation board.

sreeramIfx committed:
drivers: pinctrl: pincontrol driver updates to support PSC3

Update pincontrol driver to support PSC3

Signed-off-by: Sreeram Tatapudi <sreeram.praveen@infineon.com>

sreeramIfx committed:
drivers: serial: Adding PDL based UART driver

Adding a basic UART driver based on the PDL API

Signed-off-by: Sreeram Tatapudi <sreeram.praveen@infineon.com>

nordicjm commented on soc/infineon/cat1b/psc3/Kconfig:
```diff
@@ -0,0 +1,32 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+DT_COMPAT_ARM_CORTEX_M33 := arm,cortex-m33
+
+config SOC_SERIES_PSC3
+	select ARM
+	select CPU_CORTEX_M33 if $(dt_compat_enabled,$(DT_COMPAT_ARM_CORTEX_M33))
```
this is not right, use cpu clusters properly in soc.yml then have the correct secondary symbol here which would be based on the soc e.g. `config SOC_PSC3M5FDS2ABQ1_M33` select the `CPU_CORTEX_M33` symbol

mcatee-infineon commented on boards/infineon/kit_psc3m5_evk/Kconfig.defconfig:
```diff
@@ -0,0 +1,4 @@
+# The Infineon PSOC™ Control C3M5 Evaluation Kit (KIT_PSC3M5-EVK)
+
+# Copyright (c) 2025 Cypress Semiconductor Corporation.
```
@nordicjm This file should be removed now

mcatee-infineon commented on boards/infineon/kit_psc3m5_evk/doc/img/kit_psc3m5_evk.jpg:
@nordicjm Does it look correct now?

mcatee-infineon commented on boards/infineon/kit_psc3m5_evk/kit_psc3m5_evk.dts:
```diff
@@ -0,0 +1,81 @@
+};
+
+
+
```
@nordicjm, this should be done

mcatee-infineon commented on soc/infineon/cat1b/psc3/Kconfig:
```diff
@@ -0,0 +1,24 @@
+# Series definitions
+
+DT_COMPAT_ARM_CORTEX_M33 := arm,cortex-m33
+DT_COMPAT_ARM_CORTEX_M55 := arm,cortex-m55
```
@nordicjm, the reference to m55 should be gone from here now.

mcatee-infineon commented on soc/infineon/cat1b/psc3/Kconfig.defconfig:
```diff
@@ -0,0 +1,33 @@
+	default 180000000 if SOC_DIE_PSC3M5
+	default 180000000 if SOC_DIE_PSC3P5
+	default 100000000 if SOC_DIE_PSC3P2
+	default 100000000 if SOC_DIE_PSC3M3
```
@nordicjm, does what is there now look good to you?

mcatee-infineon commented on soc/infineon/cat1b/psc3/Kconfig.soc:
```diff
@@ -0,0 +1,298 @@
+
+config SOC_SERIES
+	default "psc3" if SOC_DIE_PSC3M3
+	default "psc3" if SOC_DIE_PSC3M5
```
@nordicjm, this should be done now.

nordicjm commented on boards/infineon/kit_psc3m5_evk/kit_psc3m5_evk.dts:
```diff
@@ -0,0 +1,78 @@
+	};
+};
+
+
```
```suggestion
```

nordicjm commented on boards/infineon/kit_psc3m5_evk/kit_psc3m5_evk_psc3m5fds2afq1_ns.dts:
```diff
@@ -0,0 +1,85 @@
+		zephyr,shell-uart = &uart3;
+	};
+};
+
```
```suggestion
```

nordicjm commented on dts/arm/infineon/cat1b/psc3/system_clocks.dtsi:
```diff
@@ -0,0 +1,112 @@
+		};
+
+	};
+
```
```suggestion
```

nordicjm commented on boards/infineon/kit_psc3m5_evk/Kconfig.kit_psc3m5_evk:
```diff
@@ -0,0 +1,7 @@
+# Copyright (c) 2025 Cypress Semiconductor Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_KIT_PSC3M5_EVK
```
This is not fixed, first commit adds dts files? And a board which uses this soc: https://github.com/zephyrproject-rtos/zephyr/pull/91971/commits/9c15907bd832dfb1420844e1fef162f5f05941db#diff-a154d3a4730e4bea0cc19bfae4ec06dde11e2c41ea0caacd4e15f43711d5f3edR6
second commit adds a random file to a board?? And soc files which were used by the first commit: https://github.com/zephyrproject-rtos/zephyr/pull/91971/commits/49b993c0818fb38f28302920dd9b47201871d891

The commit ordering needs to be fixed, the commit contents needs to be fixed e.g. one commit adding dts and soc files, one commit (per) board, etc.

nordicjm reviewed (changes requested):
https://github.com/zephyrproject-rtos/zephyr/pull/91971#discussion_r2224559683 not fixed

sreeramIfx committed:
drivers: gpio: GPIO driver updates to support PSC3

Update GPIO driver to support PSC3

Signed-off-by: Sreeram Tatapudi <sreeram.praveen@infineon.com>

sreeramIfx committed:
drivers: Implement initial version of ifx_cat1 pdl clock_control

- Shift ifx_cat1 clock_control driver to using pdl instead of hal calls
- add soc.c file containing call to SystemInit()
- Update board's dts files
- add binding for peri divs
- update system_clocks.dtsi for psc3
- add new peri clock_control driver

Signed-off-by: Sreeram Tatapudi <sreeram.praveen@infineon.com>

kartben commented on boards/infineon/kit_psc3m5_evk/doc/index.rst:
```diff
@@ -0,0 +1,122 @@
+Overview
+********
+
+This is the standard evaluation board for our PSOC™ Control C3 family of MCU's. PSOC™ Control C3M5
```
"our" should be avoided please. This is Zephyr documentation, not Infineon's :)
Please review/fix this kind of wording in the entire file

sreeramIfx committed:
boards: infineon: psc3m5_evk: Adding folder structure for PSC3M5_EVK

    * Adding folder structure and first pass config for PCS3M5_EVK
    * Adding dtsi, kconfig, and other files for KIT_PSC3M5_EVK
      evaluation board.

Signed-off-by: Sreeram Tatapudi <sreeram.praveen@infineon.com>

parthitce commented on dts/arm/infineon/cat1b/mpns/psc3m3edabq1.dtsi:
```diff
@@ -0,0 +1,44 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
```
`SPDX-License-Identifier: Apache-2.0` shall be enough. Below statement is not required. Applies in all the files.

sreeramIfx replied:
removed the additional text

parthitce commented on dts/arm/infineon/cat1b/psc3/psc3.e-lqfp-48.dtsi:
shall the fine named as pinctrl prefix ?

parthitce commented on soc/infineon/cat1b/psc3/CMakeLists.txt:
is it possible to split soc and dts into separate commits?

sreeramIfx replied:
split into separate commits

parthitce commented on soc/infineon/cat1b/psc3/Kconfig.defconfig:
```diff
@@ -0,0 +1,30 @@
+if SOC_SERIES_PSC3
+
+config NUM_IRQS
+	default 140 if CPU_CORTEX_M33
```
will there be SoC with different CPU?

parthitce commented on boards/infineon/kit_psc3m5_evk/Kconfig.kit_psc3m5_evk:
```diff
@@ -0,0 +1,7 @@
+# The Infineon PSOC™ Control C3M5 Evaluation Kit (KIT_PSC3M5_EVK)
```
fix commit message alignment

parthitce commented on modules/hal_infineon/mtb-pdl-cat1/CMakeLists.txt:
```diff
@@ -101,6 +101,8 @@ if(CONFIG_SOC_FAMILY_INFINEON_CAT1B)
   zephyr_library_sources(${pdl_drv_dir}/source/cy_syspm_v2.c)
   zephyr_library_sources(${pdl_drv_dir}/source/cy_syspm_btss.c)
   zephyr_library_sources(${pdl_drv_dir}/source/cy_syspm_ppu.c)
+  zephyr_library_sources(${pdl_drv_dir}/source/cy_ppc.c)
```
IMO this can be split into another commit as well

parthitce commented on boards/infineon/kit_psc3m5_evk/kit_psc3m5_evk_defconfig:
```diff
@@ -7,3 +7,6 @@
 CONFIG_TRUSTED_EXECUTION_SECURE=y
+
+# Enable GPIO driver
+CONFIG_GPIO=y
```
possible to combine this with board patch?

parthitce commented on drivers/gpio/gpio_ifx_cat1.c:
```diff
@@ -93,7 +93,11 @@ static int gpio_cat1_configure(const struct device *dev, gpio_pin_t pin, gpio_fl
 		return -ENOTSUP;
 	}
 
+#if defined(CY_PDL_TZ_ENABLED)
```
split into separate commit and re-order this commit to base

parthitce commented on drivers/pinctrl/pinctrl_ifx_cat1.c:
```diff
@@ -20,22 +20,17 @@
-	GPIO_PORT_OR_NULL(DT_NODELABEL(gpio_prt13)),
-	GPIO_PORT_OR_NULL(DT_NODELABEL(gpio_prt14))
-};
+	GPIO_PORT_OR_NULL(DT_NODELABEL(gpio_prt0)),  GPIO_PORT_OR_NULL(DT_NODELABEL(gpio_prt1)),
```
how about adding new ports in suffix and leaving the existing code intact ?

parthitce commented on boards/infineon/kit_psc3m5_evk/kit_psc3m5_evk.dts:
```diff
@@ -39,7 +41,7 @@ uart1: &scb3 {
 
 &path_mux1 {
-	status = "okay";
+	status = "disabled";
```
this isn't correct. either enable or disable in one patch

parthitce commented on boards/infineon/kit_psc3m5_evk/kit_psc3m5_evk.dts:
```diff
@@ -30,6 +30,8 @@ uart1: &scb3 {
 	status = "okay";
 	current-speed = <115200>;
 
+	clocks = <&peri0_group4_8bit_0>;
```
move the clock driver patch to base and this can go with board files

parthitce commented on drivers/clock_control/CMakeLists.txt:
```diff
@@ -26,6 +26,9 @@ zephyr_library_sources_ifdef(CONFIG_CLOCK_CONTROL_NRF                 clock_cont
 zephyr_library_sources_ifdef(CONFIG_CLOCK_CONTROL_NRF_DRIVER_CALIBRATION nrf_clock_calibration.c)
 zephyr_library_sources_ifdef(CONFIG_CLOCK_CONTROL_RV32M1_PCC          clock_control_rv32m1_pcc.c)
 zephyr_library_sources_ifdef(CONFIG_CLOCK_CONTROL_INFINEON_CAT1       clock_control_ifx_cat1.c)
+zephyr_library_sources_ifdef(CONFIG_CLOCK_CONTROL_IFX_CAT1_FIXED_CLOCK       clock_control_ifx_cat1_fixed_clock.c)
```
Do we need 3 different drivers? I will check the clock tree, could you please move this commit to base.

parthitce commented on boards/infineon/kit_psc3m5_evk/kit_psc3m5_evk-common.dtsi:
```diff
@@ -64,6 +64,10 @@
 	status = "okay";
 };
 
+&gpio_prt6 {
```
likewise, makes sense to keep all the baord related changes separate

parthitce reviewed (changes requested):
- Commits needs re-order
- Groups changes for SoC, dts and board

Still this PR looks heavy in-terms of added lines. But removing the License text should help IMO.

sreeramIfx committed:
dts: infineon: psc3: Adding DTS files for PSC3 series

Adding DTS files and MPN files for PCS3M5_EVK

Signed-off-by: Sreeram Tatapudi <sreeram.praveen@infineon.com>

sreeramIfx committed:
soc: infineon: psc3: Adding folder structure for PSC3 series

Adding kconfig, and other soc files for PSOC Control C3 devices

Signed-off-by: Sreeram Tatapudi <sreeram.praveen@infineon.com>

sreeramIfx commented on dts/arm/infineon/cat1b/mpns/psc3m3edabq1.dtsi:
```diff
@@ -0,0 +1,44 @@
+/*
```
updated the commit message

parthitce reviewed (approved):
LGTM
